### PR TITLE
Documentation for the Timer Driver

### PR DESCRIPTION
**Pull request by saramonteiro**
## Summary
This PR adds detailed instructions on how to use the Timer Driver.

## Impact
N/A

## Testing
N/A

### CODE REVIEW

**Comment by saramonteiro** [comment]
@v01d Please, take a look. I followed your suggestions in our talking.

File: Documentation/components/drivers/character/timer.rst [hunk 1]
```diff
@@ -16,3 +16,274 @@ locations:
...
+
+
+At an application level, the timer functionalities may be accessed through ``ioctl`` systems calls. The available ``ioctl`` commands are:
+
+* TCIOC_START
+* TCIOC_STOP
+* TCIOC_GETSTATUS
+* TCIOC_SETTIMEOUT 
+* TCIOC_NOTIFICATION
+* TCIOC_MAXTIMEOUT
```
Comment (v01d) [inline Documentation/components/drivers/character/timer.rst:92]: you can use `macro` for these

File: Documentation/components/drivers/character/timer.rst [hunk 2]
```diff
@@ -16,3 +16,274 @@ locations:
...
+
+  fd = open(devname, O_RDONLY);
+  if (fd < 0)
+    {
+      fprintf(stderr, "ERROR: Failed to open %s: %d\n",
+              devname, errno);
+      return EXIT_FAILURE;
+    }
+
+TCIOC_START
```
Comment (v01d) [inline Documentation/components/drivers/character/timer.rst:111]: you can use macro (same for every other subsection)

**Comment by v01d** [comment]
Changes look good. I added a few comments. Also, I'm not sure if it is good idea to mention internal functions such as `start`/`stop`. Or do are you refering to the lower-half interface operations? If so, maybe you can mention that explicitly and show the ops struct here

**Comment by saramonteiro** [comment]
I applied the macros and yes, I meant operations. I added now the struct, please take a look and let me know what you think. Maybe it is really getting too much into detail. Maybe We can consider deleting the ops related info

**Comment by v01d** [comment]
I think it is OK to keep that if it clears up the docs.

**Review by v01d — APPROVED** [review]

**Comment by davids5** [comment]
Nice! Where is this published too?

**Comment by v01d** [comment]
Docs are at: https://nuttx.apache.org/docs/latest/
However they are only updated daily for now, as we lack the proper setup to rebuild the website on pushes to this repo.

**Comment by davids5** [comment]
@v01d - Thank you!

**Pull request merged**